std/json.d

```diff
@@ -1499,16 +1500,16 @@ unittest
         `0`,
         `123`,
         `-4321`,
-        `0.23`,
-        `-0.23`,
+        `0.23000000417232513`,
+        `-0.23000000417232513`,
```

Comment:
> Actually %.16g is enough since 53 (fracp num bits) * 2 log10 = 15.9545897702.

Yes, 16 decimal digits should be enough. But, you need to undo the changes you made to these lines. This PR is not ready until it can pass the original tests.

To 16 digits of precision, `0.23` is the correct `double` representation of `0.23`. The values you wrote above are truncated to `float` precision, which is not right.